klusterletconfig/v1alpha1/types.go

```diff
@@ -0,0 +1,82 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=klusterletconfigs
+
+// KlusterletConfig contains the configuration of a klusterlet including the upgrade strategy, config overrides, proxy configurations etc.
+type KlusterletConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of KlusterletConfig
+	// +optional
+	Spec KlusterletConfigSpec `json:"spec,omitempty"`
+
+	// Status defines the observed state of KlusterletConfig
+	// +optional
+	Status KlusterletConfigStatus `json:"status,omitempty"`
+}
+
+// KlusterletConfigSpec defines the desired state of KlusterletConfig, usually provided by the user.
+type KlusterletConfigSpec struct {
```

Comment:
@elgnay do we define the proxy fileds in this PR too?

Reply:
It will be in a separated PR.